script.py

```diff
@@ -0,0 +1,129 @@
+import requests
+import csv
+import json
+import os
+import sys
+from urllib.parse import urlparse
+
+OBJECT_SPECS = dict(
+    checkouts=dict(
+        patchset_files=[dict(url=True)],
+    ),
+    builds=dict(
+        input_files=[dict(url=True)],
+        output_files=[dict(url=True)],
+        config_url=True,
+        log_url=True,
+    ),
+    tests=dict(
+        output_files=[dict(url=True)],
+        log_url=True,
+    ),
+)
+
+def extract_fields(spec, data):
+    """
+    Extract values of fields from data according to a specification.
+
+    Args:
+        spec:   The specification of fields to extract the values from.
+        data:   The data to extract the field values from.
+
+    Returns:
+        An array of tuples, where each tuple contains:
+            * A tuple containing dictionary keys leading to the field
+            * The field value
+    """
+    tuples = []
+
+    if isinstance(spec, dict):
+        for k, v in spec.items():
+            if k in data:
+                for t in extract_fields(v, data[k]):
+                    tuples.append(((k,) + t[0], t[1]))
+    elif isinstance(spec, list):
+        assert len(spec) == 1
+        for idx, d in enumerate(data):
+            tuples += extract_fields(spec[0], d)
+    elif spec == True:
+        return [(tuple(), data)]
+
+    return tuples
+
+# Prepare CSV file
+csv_writer = csv.writer(sys.stdout)
+csv_writer.writerow([
+    'Origin', 'Field Path', 'URL', 'Status Code',
+    'Content Type', 'File Extension', 'URL Length (Bytes)', 'Size'
+])
+
+URL_INFO_CACHE = {}
+URL_PATH_SET = set()  # Track unique field path and URL combinations
+
+# Helper function to extract URL and size
+def get_url_info(url):
+    if url not in URL_INFO_CACHE:
+        try:
+            if url.endswith('/'):
```

Comment:
This is specific to the results we get. Some CI systems send links to index pages, instead of actual files, which is against the schema, and we don't really want to cache them (and we can't really cache the files listed on them).

Reply:
This of course skews our data, but :shrug:

Reply:
I see! Let's add a comment there, then, Abhishek ;)

Reply:
where?

Reply:
Just to bring the Slack's discussion.

> Did u mean  why we added /
> < Yes
> Got it

Additionally, I wrote what I meant in other comment

Reply:
okay